perses/rjmc/geometry.py

```diff
@@ -2300,9 +2472,27 @@ def __init__(self, topology_proposal, direction="forward"):
             raise Exception(msg)
 
         # Choose the first of the new atoms to find the corresponding residue:
-        transforming_residue = self._new_atom_objects[self._new_atoms[0]].residue
+        #transforming_residue = self._new_atom_objects[self._new_atoms[0]].residue
+
+        self._residue_graph = _nx_graph.graph
+        self._reference_connectivity_graph = self._create_reference_connectivity_graph()
+
+    def _create_reference_connectivity_graph(self):
```

Comment:
we generate a reference connectivity graph with reference to the old networkx representation of the old oemol; when we add new atoms, we check whether rings are closing; if this is the case, we add bonds across torsions that are not being used to perform atom placement into the `omitted_terms` dict